pkg/api/aim/query/query.go

```diff
@@ -733,13 +733,23 @@ func (pq *parsedQuery) parseUnaryOp(node *ast.UnaryOp) (any, error) {
 	if err != nil {
 		return nil, err
 	}
-	c, ok := e.(clause.Expression)
-	if !ok {
-		return nil, fmt.Errorf("not a valid SQL expression: %#v", e)
-	}
 	switch node.Op {
+	case ast.USub:
+		switch e := e.(type) {
+		case int:
+			return -e, nil
+		case float64:
+			return -e, nil
+		default:
+			return nil, fmt.Errorf("unsupported unary operation %q on %T", node.Op, e)
```

Comment:
maybe `unsupported` type?

Reply:
good point!
addressed in 5005490d6375e9d6856325dfb12d7b7cd29cd866